drivers/sensor/fdc2x1x/fdc2x1x.c

```diff
@@ -0,0 +1,1002 @@
+	if (ret) {
+		return ret;
+	}
+
+	ret = fdc2x1x_set_deglitch(dev, cfg->deglitch);
+	if (ret) {
+		return ret;
+	}
+
+	if (!data->fdc221x) {
+		ret = fdc2x1x_set_output_gain(dev, cfg->output_gain);
+		if (ret) {
+			return ret;
+		}
+	}
+
+	ret = fdc2x1x_set_active_channel(dev, cfg->active_channel);
+	if (ret) {
+		return ret;
+	}
+
+	ret = fdc2x1x_set_sensor_activate_sel(dev, cfg->sensor_activate_sel);
+	if (ret) {
+		return ret;
+	}
+
+	ret = fdc2x1x_set_ref_clk_src(dev, cfg->clk_src);
+	if (ret) {
+		return ret;
+	}
+
+#ifdef CONFIG_FDC2X1X_TRIGGER_NONE
+	/* Enable Data Ready Flag to poll for new measurement */
+	ret = fdc2x1x_reg_write_mask(dev, FDC2X1X_ERROR_CONFIG,
+				     FDC2X1X_ERROR_CONFIG_DRDY_2INT_MSK,
+				     FDC2X1X_ERROR_CONFIG_DRDY_2INT_SET(1));
+	if (ret) {
+		return ret;
+	}
+
+	/* INTB asserts by default, so disable it here */
+	ret = fdc2x1x_set_interrupt_pin(dev, false);
+	if (ret) {
+		return ret;
+	}
+#endif
+
+	ret = fdc2x1x_set_current_drv(dev, cfg->current_drv);
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+/**
+ * Probe device (Check if it is the correct device).
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int fdc2x1x_probe(const struct device *dev)
+{
+	struct fdc2x1x_data *data = dev->data;
+	uint16_t dev_id, man_id;
+
+	if (fdc2x1x_reg_read(dev, FDC2X1X_DEVICE_ID, &dev_id) < 0) {
+		LOG_ERR("Failed to read device id");
+		return -EIO;
+	}
+
+	if (dev_id == FDC2X1X_DEVICE_ID_VAL_28BIT) {
+		data->fdc221x = true;
+	} else if (dev_id == FDC2X1X_DEVICE_ID_VAL) {
+		data->fdc221x = false;
+	} else {
+		LOG_ERR("Wrong device id");
+		return -ENODEV;
+	}
+
+	if (fdc2x1x_reg_read(dev, FDC2X1X_MANUFACTURER_ID, &man_id) < 0) {
+		LOG_ERR("Failed to read manufacturer id");
+		return -EIO;
+	}
+
+	if (man_id != FDC2X1X_MANUFACTURER_ID_VAL) {
+		LOG_ERR("Wrong manufaturer id");
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+/**
+ * Initialize the SD-Pin.
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int fdc2x1x_init_sd_pin(const struct device *dev)
+{
+	struct fdc2x1x_data *data = dev->data;
+	const struct fdc2x1x_config *cfg = dev->config;
+
+	data->gpio = device_get_binding(cfg->sd_name);
+	if (!data->gpio) {
+		LOG_ERR("Failed to get pointer to %s device!", cfg->sd_name);
+		return -EINVAL;
+	}
+
+	gpio_pin_configure(data->gpio, cfg->sd_pin,
+			   GPIO_OUTPUT_INACTIVE | cfg->sd_flags);
+
+	return 0;
+}
+
+/**
+ * Initialization of the device.
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int fdc2x1x_init(const struct device *dev)
+{
+	struct fdc2x1x_data *data = dev->data;
+	const struct fdc2x1x_config *cfg = dev->config;
+	uint8_t ch_supported;
+
+	if (cfg->fdc2x14) {
+		ch_supported = 4;
+	} else {
+		ch_supported = 2;
+	}
+
+	if (cfg->num_channels <= 0) {
```

Comment:
Taking another look at this line, it's using `<= 0` on an unsigned type. That's a bit misleading, please use `== 0`.

Reply:
Ok, now fixed